tests/components/mealie/test_todo.py

```diff
@@ -0,0 +1,171 @@
+"""Tests for the Mealie todo."""
+
+from unittest.mock import AsyncMock, patch
+
+from aiomealie.exceptions import MealieError
+import pytest
+from syrupy.assertion import SnapshotAssertion
+
+from homeassistant.components.todo import DOMAIN as TODO_DOMAIN
+from homeassistant.const import ATTR_ENTITY_ID, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers import entity_registry as er
+
+from . import setup_integration
+
+from tests.common import MockConfigEntry, snapshot_platform
+
+
+async def test_entities(
+    hass: HomeAssistant,
+    snapshot: SnapshotAssertion,
+    entity_registry: er.EntityRegistry,
+    mock_mealie_client: AsyncMock,
+    mock_config_entry: MockConfigEntry,
+) -> None:
+    """Test todo entities."""
+    with patch("homeassistant.components.mealie.PLATFORMS", [Platform.TODO]):
+        await setup_integration(hass, mock_config_entry)
+
+    await snapshot_platform(hass, entity_registry, snapshot, mock_config_entry.entry_id)
+
+
+async def test_get_todo_list_items(
+    hass: HomeAssistant,
+    snapshot: SnapshotAssertion,
+    mock_mealie_client: AsyncMock,
+    mock_config_entry: MockConfigEntry,
+) -> None:
+    """Test for getting a To-do list."""
+    await setup_integration(hass, mock_config_entry)
+
+    state = hass.states.get("todo.mealie_supermarket")
+    assert state
+    assert state == snapshot
```

Comment:
This is now already tested via the snapshot test above so we could remove this imo. Please don't forget to delete the snapshots and regenerate them